shor/layers.py

```diff
@@ -16,7 +16,7 @@ class Qbits(_Layer, Iterable):
     def __init__(self, num, state=0, **kwargs):
         self.num = num
         self.state = state
-        self._qbits = list(range(num))
+        self._qbits = [(i, self) for i in range(num)]
```

Comment:
This looks like the right change for qbits (A lot of unit tests will probably break though lol)

Reply:
list(zip(range(num),[self]*num))  this is something you had before I think--is this better?